### PR TITLE
hw: m25p80: upstream suggestions for block protect

### PR DESCRIPTION
**Pull request by chen-iris**
Signed-off-by: Iris Chen <irischenlj@fb.com>

### CODE REVIEW

File: hw/block/m25p80.c [hunk 1]
```diff
@@ -631,31 +631,33 @@ void flash_write8(Flash *s, uint32_t addr, uint8_t data)
     uint32_t page = addr / s->pi->page_size;
     uint8_t prev = s->storage[s->cur_addr];
 
-    if (!s->write_enable) {
-        qemu_log_mask(LOG_GUEST_ERROR, "M25P80: write with write protect!\n");
-        return;
-    }
```
Comment (peterdelevoryas) [inline hw/block/m25p80.c:637]: How come this was moved?

**Review by peterdelevoryas — CHANGES_REQUESTED** [review]

File: hw/block/m25p80.c (hunk 1 above)
Comment (chen-iris) [reply hw/block/m25p80.c:637]: Upstream review stipulates that the declarations should be at the top of the function. So, rather, I moved the block_protect_value declaration above this

**Comment by peterdelevoryas** [comment]
On Wed, Jul 06, 2022 at 09:15:51AM -0700, Iris Chen wrote:
> @chen-iris commented on this pull request.
> 
> 
> 
> > -    if (!s->write_enable) {
> -        qemu_log_mask(LOG_GUEST_ERROR, "M25P80: write with write protect!\n");
> -        return;
> -    }
> 
> Upstream review stipulates that the declarations should be at the top of the function. So, rather, I moved the block_protect_value declaration above this

Ohhh I understand now. Yeah that's fine then.

**Comment by chen-iris** [comment]
> On Wed, Jul 06, 2022 at 09:15:51AM -0700, Iris Chen wrote: @chen-iris commented on this pull request. > - if (!s->write_enable) { - qemu_log_mask(LOG_GUEST_ERROR, "M25P80: write with write protect!\n"); - return; - } Upstream review stipulates that the declarations should be at the top of the function. So, rather, I moved the block_protect_value declaration above this
> Ohhh I understand now. Yeah that's fine then.
> […](#)
> -- Reply to this email directly or view it on GitHub: [#3 (comment)](https://github.com/facebook/openbmc-qemu/pull/3#discussion_r915027759) You are receiving this because you commented. Message ID: ***@***.***>

Do you have any other comments -- does the m25p80_block_protect_needed function look alright

**Comment by peterdelevoryas** [comment]
> > On Wed, Jul 06, 2022 at 09:15:51AM -0700, Iris Chen wrote: @chen-iris commented on this pull request. > - if (!s->write_enable) { - qemu_log_mask(LOG_GUEST_ERROR, "M25P80: write with write protect!\n"); - return; - } Upstream review stipulates that the declarations should be at the top of the function. So, rather, I moved the block_protect_value declaration above this
> > Ohhh I understand now. Yeah that's fine then.
> > […](#)
> > -- Reply to this email directly or view it on GitHub: [#3 (comment)](https://github.com/facebook/openbmc-qemu/pull/3#discussion_r915027759) You are receiving this because you commented. Message ID: _**@**_.***>
> 
> Do you have any other comments -- does the m25p80_block_protect_needed function look alright

I think it looks ok yeah.

**Review by peterdelevoryas — APPROVED** [review]

**Pull request merged**